modules/imgproc/src/hal_replacement.hpp

```diff
@@ -630,6 +630,38 @@ inline int hal_ni_medianBlur(const uchar* src_data, size_t src_step, uchar* dst_
 #define cv_hal_medianBlur hal_ni_medianBlur
 //! @endcond
 
+/**
+   @brief Calculates adaptive threshold
+   @param src_data,src_step Source image
+   @param dst_data,dst_step Destination image
+   @param width,height Source image dimensions
+   @param maxValue Value assigned to the pixels for which the condition is satisfied
+   @param adaptiveMethod Adaptive thresholding algorithm
+   @param thresholdType Thresholding type
+   @param blockSize Size of a pixel neighborhood that is used to calculate a threshold value
+   @param C Constant subtracted from the mean or weighted mean
+*/
+inline int hal_ni_adaptiveThreshold(const uchar* src_data, size_t src_step, uchar* dst_data, size_t dst_step, int width, int height, double maxValue, int adaptiveMethod, int thresholdType, int blockSize, double C) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
```

Comment:
You should define allowed values for _adaptiveMethod_ and _thresholdType_ in the [opencv/modules/imgproc/include/opencv2/imgproc/hal/interface.h](https://github.com/opencv/opencv/blob/master/modules/imgproc/include/opencv2/imgproc/hal/interface.h).